src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Configuration;
+
+namespace NuGet.PackageManagement.UI
+{
+    internal static class PackageSourceMappingUtility
+    {
+        internal static void ConfigureNewPackageSourceMapping(
```

Comment:
The reason I didn't put these UI related concepts into the `PackageSourceMappingProvider` is to keep a separation of concerns and to minimize the public API changes. I wouldn't want that provider knowing about things like `UserAction`.

It may be that the provider can have additional logic later, but I don't want to create public APIs until we know exactly what that provider's surface should look like.